broadlink/device.py

```diff
@@ -2,13 +2,93 @@
 import threading
 import random
 import time
-from typing import Tuple, Union
+from datetime import datetime
+from typing import Generator, Tuple, Union
 
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
 
 from .exceptions import check_error, exception
 
+HelloResponse = Tuple[int, Tuple[str, int], str, str, bool]
+
+
+def scan(
+    timeout: int = 10,
+    local_ip_address: str = None,
+    discover_ip_address: str = '255.255.255.255',
+    discover_ip_port: int = 80,
+) -> Generator[HelloResponse, None, None]:
+    """Broadcast a hello message and yield responses."""
+    cs = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    cs.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    cs.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+
+    if local_ip_address:
+        cs.bind((local_ip_address, 0))
```

Comment:
This line is very aggressive. It is also unnecessary, so I'm gonna make it optional.